src/ngx_http_encrypted_session_cipher.c

```diff
@@ -19,14 +19,23 @@
 static uint64_t ngx_http_encrypted_session_ntohll(uint64_t n);
 static uint64_t ngx_http_encrypted_session_htonll(uint64_t n);
 
+typedef int (*cipher_ctx_reset_handle) (EVP_CIPHER_CTX *ctx);
+
+#if OPENSSL_VERSION_NUMBER >= 0x1010000fL
+static cipher_ctx_reset_handle reset_cipher_ctx = EVP_CIPHER_CTX_reset;
+#else
+static cipher_ctx_reset_handle reset_cipher_ctx = EVP_CIPHER_CTX_cleanup;
```

Comment:
Can we put these into the main conf and register a pool cleanup handle for it instead? Otherwise it's not valgrind clean and may have races upon HUP reload, maybe?